cmake/FindnetCDF.cmake

```diff
@@ -28,11 +28,7 @@
 include(BOUT++functions)
 include(CMakePrintHelpers)
 
-if (BOUT_NETCDF_VERSION)
```

Comment:
It looks like this might be a case where someone wants to specify a particular version. Is there a way to achieve this without the EXACT flag to find_package?

Reply:
Hmm, I thought that `netCDF_FIND_VERSION` would basically do this out of the box, but it seems it might not work when it looks for `CONFIG` files.

The other, out of the box, method is to to use `netCDF_ROOT=/path/to/specific/version`